packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -466,23 +469,32 @@ export class ExecutionEngineHttp implements IExecutionEngine {
     return response.map(deserializeExecutionPayloadBody);
   }
 
-  async getBlobs(_fork: ForkName, versionedHashes: VersionedHashes): Promise<(BlobAndProof | null)[]> {
+  async getBlobs(fork: ForkPostFulu, versionedHashes: VersionedHashes): Promise<BlobAndProofV2[]>;
+  async getBlobs(fork: ForkPreFulu, versionedHashes: VersionedHashes): Promise<(BlobAndProof | null)[]>;
+  async getBlobs(
+    fork: ForkName,
+    versionedHashes: VersionedHashes
+  ): Promise<BlobAndProofV2[] | (BlobAndProof | null)[]> {
```

Comment:
the type here hasn't really changed so we can keep blob and proof

also the return type should be 
`Promise<BlobAndProofV2[] | null | (BlobAndProof | null)[]>`

see https://discord.com/channels/595666850260713488/1252403418941624532/1359818035229360199

i think we should just separate getblobsv2 call because the context is very different and it will be used in a very different way in the fetching calls for the peerdas. no point confusing the same signature as the context has widely changed

Reply:
Updated to handle null returns from `engine_getblobsv2`. The type now contains a `proofs` array of cell proofs that replaces `proof`, a single blob proof, so the values of those aren't necessarily interchangeable.

Discussion on separating getblobsv2 happening in this PR: https://github.com/ChainSafe/lodestar/pull/7675